autogpt_platform/backend/backend/server/v2/chat/routes.py

```diff
@@ -0,0 +1,234 @@
+"""Chat API routes for chat session management and streaming via SSE."""
+
+import logging
+from collections.abc import AsyncGenerator
+from typing import Annotated
+
+from autogpt_libs import auth
+from fastapi import APIRouter, Depends, Query, Security
+from fastapi.responses import StreamingResponse
+from fastapi.security import HTTPAuthorizationCredentials, HTTPBearer
+from pydantic import BaseModel
+
+import backend.server.v2.chat.service as chat_service
+from backend.server.v2.chat.config import ChatConfig
+from backend.util.exceptions import NotFoundError
+
+config = ChatConfig()
+
+
+logger = logging.getLogger(__name__)
+
+# Optional bearer token authentication
+optional_bearer = HTTPBearer(auto_error=False)
+
+router = APIRouter(
+    tags=["chat"],
+)
+
+
+def get_optional_user_id(
+    credentials: HTTPAuthorizationCredentials | None = Security(optional_bearer),
+) -> str | None:
+    """Extracts the user ID from a bearer JWT if present, otherwise returns None for anonymous access."""
+    if not credentials:
+        return None
+
+    try:
+        # Parse JWT token to get user ID
+        from autogpt_libs.auth.jwt_utils import parse_jwt_token
+
+        payload = parse_jwt_token(credentials.credentials)
+        return payload.get("sub")
+    except Exception as e:
+        logger.debug(f"Auth token validation failed (anonymous access): {e}")
+        return None
+
+
+# ========== Request/Response Models ==========
+
+
+class CreateSessionResponse(BaseModel):
+    """Response model containing information on a newly created chat session."""
+
+    id: str
+    created_at: str
+    user_id: str | None
+
+
+class SessionDetailResponse(BaseModel):
+    """Response model providing complete details for a chat session, including messages."""
+
+    id: str
+    created_at: str
+    updated_at: str
+    user_id: str | None
+    messages: list[dict]
+
+
+# ========== Routes ==========
+
+
+@router.post(
+    "/sessions",
+)
+async def create_session(
+    user_id: Annotated[str | None, Depends(get_optional_user_id)],
```

Comment:
so anyone can chat even non logged in user? won't this be very prone to denial of service ?